scalajvm/app/controllers/SuggestedQuotes.scala

```diff
@@ -5,27 +5,31 @@ import models.forms.SuggestQuoteForm
 import play.api.data.Forms._
 import play.api.data._
 import play.api.mvc._
+import play.api.i18n
+import play.api.Configuration
 import services.SuggestedQuoteService
+import javax.inject._
 
 import scala.concurrent.Await
 import scala.concurrent.duration._
 
-object SuggestedQuotes extends Controller {
-  def newQuote() = Action { request =>
+@Singleton
+class SuggestedQuotes @Inject()(implicit cc: ControllerComponents, configuration: Configuration) extends AbstractController(cc) with i18n.I18nSupport {
+  def newQuote() = Action { implicit request =>
     Ok(views.html.newQuote(quoteForm))
   }
 
-  def addQuote() = ActionWithTx { implicit request =>
+  def addQuote() = (new ActionWithTx(cc)) { implicit request =>
     import request.dbSession
 
-    quoteForm.bindFromRequest.fold(
+    quoteForm.bindFromRequest().fold(
 
       formWithErrors => {
         BadRequest(views.html.newQuote(formWithErrors))
       },
       form => {
         val remoteAddress = request.remoteAddress
-        if (Await.result(ReCaptcha.check(remoteAddress, form.reCapthaResponse), 30.seconds)) { // TODO: Async check
+        if (Await.result((new ReCaptcha).check(remoteAddress, form.reCapthaResponse), 30.seconds)) { // TODO: Async check
```

Comment:
You've turned `ReCaptcha` into an injectable component, but still all it via `new`. Shoulnd't you inject it here, then?

Reply:
The original intent behind adding `@Inject`s was to get access to Play! components, since the framework now uses DI for everything. Since `ReCaptcha` is a singleton, your suggestion makes perfect sense, so I addressed that in 04e9970ee0994cdafafb046e9b7b6e4f0d1b930e.